### DISCUSSION TITLE
Is it possible to create a registration and login page?

### DISCUSSION THREAD

**Discussion by Nurtilekk** [nextauthjs/next-auth#2791] [Help] [answered]
I`m using Credentials provider, i have login page and registration page, but when im adding 2 credentials provider (because i have different endpoints for register and login) next-auth sees only the last one. 
![image](https://user-images.githubusercontent.com/62701301/133925401-caae495b-a5b5-4b9c-9f8e-9b733839d5af.png)
![image](https://user-images.githubusercontent.com/62701301/133925410-6effc33b-cdc2-43bd-8b54-fd9ebe83ea7c.png)
Only the last one is shown "Register"

**Comment by Nurtilekk** [comment]
How can i implement registration and login logic? Any example?

**Comment by balazsorban44** [comment, marked as answer]
You will need a different `id` for the two providers. Since you are not defining one, both default to `"credentials"` and thus only showing one of them.

Say you add `id: "register"` to one and `id: "login"` to the other. You would then do `signIn("register")` or `signIn("login")` in your client code.

See: https://next-auth.js.org/configuration/providers/credentials-provider#options

As a suggestion, if your backend is in API routes, don't use a fetch, just import the logic of login and registration for a minor performance boost.

**Reply by Nurtilekk** [reply]
You made my day, bro :*
